### PR TITLE
Add batch normalization 

### PR DESCRIPTION
**Pull request by rola93**
The original paper uses a weigth normalization layer. It is not available on keras. However, batch normalization is. In fact, it is already in comments on the code.

This PR allows to optionaly use BatchNorm layer as a proxy for weigth normalization.

As an advantage: it is optional, and it is still compatible with previous versions of current library

As a disadvantage, it is not weigth normalization yet

### CODE REVIEW

**Comment by philipperemy** [comment]
Looks good to me. Will have to find some time to test it before merging.
